ee/tabby-webserver/src/db.rs

```diff
@@ -0,0 +1,132 @@
+use std::{env, path::PathBuf, sync::Arc};
+
+use anyhow::Result;
+use lazy_static::lazy_static;
+use rusqlite::params;
+use rusqlite_migration::{AsyncMigrations, M};
+use tokio_rusqlite::Connection;
+
+lazy_static! {
+    static ref MIGRATIONS: AsyncMigrations = AsyncMigrations::new(vec![M::up(
+        r#"
+            CREATE TABLE IF NOT EXISTS register_token (
+                id INTEGER PRIMARY KEY AUTOINCREMENT,
+                token VARCHAR(255) NOT NULL,
+                created_at TIMESTAMP DEFAULT (DATETIME('now')),
+                updated_at TIMESTAMP DEFAULT (DATETIME('now')),
+                CONSTRAINT `idx_token` UNIQUE (`token`)
+            );
+        "#
+    ),]);
+    static ref TABBY_ROOT: PathBuf = match env::var("TABBY_ROOT") {
+        Ok(x) => PathBuf::from(x),
+        Err(_) => PathBuf::from(env::var("HOME").unwrap()).join(".tabby"),
+    };
+}
+
+fn db_file() -> PathBuf {
+    TABBY_ROOT.join("db.sqlite3")
```

Comment:
I suggest make [tabby_root()](https://github.com/TabbyML/tabby/blob/76679bc2492fbcf8a5b9132f32efa2ae4480580b/crates/tabby-common/src/path.rs#L21C1-L21C1) public and use `tabby_root().join(...)` here